### ISSUE TITLE
git-ftp push fails

### ISSUE THREAD

**Issue by TimSparrow** [git-ftp/git-ftp#456]
### Bug 

* How to reproduce the issue?
issue the following command:
git-ftp --scope=stage push

* actual result
> Unknown SHA1 object, make sure you are deploying the right branch and it is up-to-date.
> Do you want to ignore and upload all files again? [y/N]: N
> Aborting...

.git-ftp.log on the remote server is updated, no files uploaded
If I press Y, I get another error:
> fatal: empty string is not a valid pathspec. please use . instead if you meant to match all paths

* expected result
upload of 2-5 files changed in the current revision

* Can you point to a Git repository to reproduce it?
No, restricted by NDA.

* Or can you describe the directory structure?

* affects 2 different repos having similar file structure (CodeIgniter 1.xx based projects)
application
system

with a set of nested subdirectories and files

* branches:
- master, v1 - both point to the current stable version of the projects. Not maintained by any automated scripts
- stage (configured by a set of shell scripts to be auto-deployed to a staging server), point to the current development version (aka v2.0)
- dev (for developers)
- other branches reserved for certain features or used by individual developers, not deployed to external servers
- 

### Environment

Please copy the output of the following commands.

 echo "$SHELL"
/bin/bash
$ uname -a
Linux <hostname> 4.15.0-24-generic #26-Ubuntu SMP Wed Jun 13 08:44:47 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
$ git-ftp version
git-ftp version 1.3.1

```

**Comment by mkllnk** [comment]
Thank you for the report. It sounds like an incompatibility with newer Git versions that was fixed in Git-ftp 1.5.0. Please try a newer version from https://launchpad.net/~git-ftp/+archive/ubuntu/ppa/.

**Comment by TimSparrow** [comment]
Thanks for the suggestion, added the ppa and updated.

Now it fails to connect to the server:
fatal: Can't access remote 'ftpes://probegin%40hostingdirect.nl:***@hostingdirect.nl', exiting...

FileZilla opened in parallel window connects successfully. Should I change anything in config?

**Comment by TimSparrow** [comment]
Push shows the following message:
**fatal: Could not get last commit. Network down? Wrong URL? Use 'git ftp init' for the initial push., exiting...**

**Comment by mkllnk** [comment]
Ah, sorry, the solution is not easy. Have a look at https://github.com/git-ftp/git-ftp/issues/452#issuecomment-395937387.

**Comment by TimSparrow** [comment]
Used to work with current curl until git was updated

**Comment by TimSparrow** [comment]
to verify, I am sending my curl -V, hope it helps

$ curl -V
curl 7.58.0 (x86_64-pc-linux-gnu) libcurl/7.58.0 OpenSSL/1.1.0g zlib/1.2.11 libidn2/2.0.4 libpsl/0.19.1 (+libidn2/2.0.4) nghttp2/1.30.0 librtmp/2.3
Release-Date: 2018-01-24
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtmp rtsp smb smbs smtp smtps telnet tftp 
Features: AsynchDNS IDN IPv6 Largefile GSS-API Kerberos SPNEGO NTLM NTLM_WB SSL libz TLS-SRP HTTP2 UnixSockets HTTPS-proxy PSL

**Comment by TimSparrow** [comment]
Rebuilt curl manually. Did not help:

git-ftp --scope=stage init
fatal: Can't access remote 'ftpes://stagingit:***@staging.internettoday.nl', exiting...

$ curl -V
curl 7.58.0 (x86_64-pc-linux-gnu) libcurl/7.58.0 OpenSSL/1.1.0g zlib/1.2.11 libidn2/2.0.4 libpsl/0.19.1 (+libidn2/2.0.4) libssh2/1.8.0 nghttp2/1.30.0 librtmp/2.3
Release-Date: 2018-01-24
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtmp rtsp scp **sftp** smb smbs smtp smtps telnet tftp 
Features: AsynchDNS IDN IPv6 Largefile NTLM NTLM_WB SSL libz TLS-SRP HTTP2 UnixSockets HTTPS-proxy PSL

**Comment by TimSparrow** [comment]
Communications error in previous commit is caused by server certificate issues.

**Comment by atefBB** [comment]
This [article](http://andrewberls.com/blog/post/adding-sftp-support-to-curl) helps me alot.

**Comment by LukasFritzeDev** [comment]
@atefBB Yes. That’s the reason why we added [this to the ReadMe file](https://github.com/git-ftp/git-ftp/blob/master/INSTALL.md#sftp-on-macos) 😉 Do you use macOS, too?

**Comment by atefBB** [comment]
I'm using Ubuntu 18.04 !

**Comment by LukasFritzeDev** [comment]
Then maybe we should change the heading for this section.
